atmos_model.F90

```diff
@@ -2390,217 +2390,220 @@ subroutine setup_exportdata(rc)
+              call block_atmos_copy(datar82d, Atm(mygrid)%u_srf, Atm_block, nb, rc=localrc)
             case ('v_srf')
-              call block_atmos_copy(datar82d, Atm(mygrid)%v_srf, Atm_block, nb)
+              call block_atmos_copy(datar82d, Atm(mygrid)%v_srf, Atm_block, nb, rc=localrc)
             case ('slmsk')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%slmsk, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%slmsk, Atm_block, nb, rc=localrc)
             case ('weasd')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%weasd, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%weasd, Atm_block, nb, rc=localrc)
             case ('tsea')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%tsfco, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%tsfco, Atm_block, nb, rc=localrc)
             case ('vtype')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vtype, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vtype, Atm_block, nb, rc=localrc)
             case ('stype')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%stype, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%stype, Atm_block, nb, rc=localrc)
             case ('vfrac')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vfrac, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vfrac, Atm_block, nb, rc=localrc)
             case ('stc')
-              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%stc, Atm_block, nb)
+              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%stc, Atm_block, nb, rc=localrc)
             case ('smc')
-              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%smc, Atm_block, nb)
+              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%smc, Atm_block, nb, rc=localrc)
             case ('snwdph')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%snowd, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%snowd, Atm_block, nb, rc=localrc)
             case ('f10m')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%f10m, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%f10m, Atm_block, nb, rc=localrc)
             case ('zorl')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%zorl, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%zorl, Atm_block, nb, rc=localrc)
             case ('t2m')
-              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%t2m, Atm_block, nb)
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%t2m, Atm_block, nb, rc=localrc)
+            case default
+              localrc = ESMF_SUCCESS
```

Comment:
The default case should be an error, not success.

Reply:
@DusanJovic-NOAA - setup_exportdata() is not comprehensive. For instance, exported fields used by the chemistry model are not included in this subroutine and should not be at this time since thy are exported at the end of FV3 run phase 1. 
`setup_exportdata()` is not used during FV3 initialization and by `update_atmos_model_state()`, which is called in run phase 2, except when FV3 is coupled with chemistry.

I understand the motivation, but setting the default case to be an error would break the coupled model.
Again, refactoring this code should not be part of this PR.

Reply:
`setup_exportdata` is not used if `cplchem` is true, so the fact that not all chemistry fields are exported is irrelevant.

Why would it break the coupled model? Are there still exported fields (other than the 'chemistry' fields) that are not handled by the 'select case'?

Reply:
`setup_exportdata()` is called by `InitializeRealize()` (`fv3_cap.F90`) even if `cplchm` is `.true.`, and this is where all those issues are coming from.

If `setup_exportdata()` was not intended to be used with the coupled chemistry model, I'd be happy to disable it when `cplchm` is set to `.true.`. Please let me know.

Reply:
I do not know if you need to call `setup_exportdata` with the chemistry model from `InitializeRealize` or not. But if you do then add all exported fields, and set the default case to error. If the default case is not an error there's no way to catch typos. For example, if instead:
`            case ('o3mr')`
I write:
`            case ('o3nr')`
model will silently continue to run, without me noticing that o3mr is not exported.

Reply:
`setup_exportdata()` is not required at all when coupling with the chemistry model. I do agree with you that, generally speaking, we should return an error when a mismatch occurs.

I suggest: 
1. checking if `cplchm` is `.true.` inside `setup_exportdata()` and bailing out if so (this is needed since `GFS_control%cplchm` is not accessible in the FV3 cap)
2. returning with an error in the default case, as you suggest.

Would this work?

Reply:
I think it should.

Reply:
@DusanJovic-NOAA - I've implemented the changes above. Please take a look.